gossip/src/cluster_info.rs

```diff
@@ -965,6 +965,21 @@ impl ClusterInfo {
         }
     }
 
+    pub fn push_restart_last_voted_fork_slots(&self, update: &[Slot], last_vote_bankhash: Hash) {
+        let now = timestamp();
+        let mut last_voted_fork_slots = RestartLastVotedForkSlots::new(
+            self.id(),
+            now,
+            last_vote_bankhash,
+            self.my_shred_version(),
+        );
+        last_voted_fork_slots.fill(update);
```

Comment:
why do we need this 2 stage initialization?
Can't `new` fill out the slots as well?

also, the update logic is quite different from `push_epoch_slots`; i.e the existing value in gossip crds is entirely discarded. is that intentional? if so then `RestartLastVotedForkSlots` api could be quite simpler (e.g. no need for 2 stage initialization).

Reply:
Now we only have one packet per validator, yes the value with newer timestamp should always replace the old one.

The 2 stage initialization was still there because we need self.max_compressed_slot_size(), which can't be called in a constructor. But maybe we can approximate assuming header etc is around 100+?

Reply:
In this initialization, `max_compressed_slot_size` is always called on a `RestartLastVotedForkSlots` with empty `slots`. So it is a constant number. You can define it as a `const` and then define a test which validates the `const` value. Similar to:
https://github.com/solana-labs/solana/blob/09e858d93/gossip/src/cluster_info.rs#L3442-L3448

Also looking at the code, when there are too many slots, `fill` is taking the initial set of slots and ignoring the remaining ones.
You probably want to do the opposite; i.e. take the last slots, not the first slots.

Reply:
Good point, actually I don't care too much about time complexity here, because it is only called once on start. So I can try to pack in as many slots as possible into one packet. Changed.